src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineView.xaml.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Windows;
+using System.Windows.Controls;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.LanguageServices.Implementation;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    /// <summary>
+    /// Interaction logic for DocumentOutlineView.xaml
+    /// </summary>
+    internal partial class DocumentOutlineView : UserControl, IVsCodeWindowEvents, IDisposable
+    {
+        private readonly DocumentOutlineViewModel _viewModel;
+        private readonly VisualStudioCodeWindowInfoService _visualStudioCodeWindowInfoService;
+        private readonly IVsEditorAdaptersFactoryService _editorAdaptersFactoryService;
+        private readonly Dictionary<IVsTextView, ITextView> _trackedTextViews = new();
```

Comment:
can you doc the threading requirements around this mutable type?

Reply:
I can, this View has the same threading requirements as all of the WPF views in roslyn: everything is always on the UI thread. It only responds to WPF events and processing them always happens on the UI thread.